editor/editor_settings.cpp

```diff
@@ -642,6 +642,8 @@ void EditorSettings::_load_defaults(Ref<ConfigFile> p_extra_config) {
 	_initial_set("text_editor/behavior/navigation/drag_and_drop_selection", true);
 	_initial_set("text_editor/behavior/navigation/stay_in_script_editor_on_node_selected", true);
 	_initial_set("text_editor/behavior/navigation/open_script_when_connecting_signal_to_existing_method", true);
+	_initial_set("text_editor/behavior/navigation/use_custom_word_separators", false);
+	_initial_set("text_editor/behavior/navigation/word_separators", "`~!@#$%^&*()-=+[{]}|;:'\",.<>/?"); // custom word separators
```

Comment:
```suggestion
	_initial_set("text_editor/behavior/navigation/word_separators", "`~!@#$%^&*()-=+[{]}|;:'\",.<>/?"); // Custom word separators.
```

Reply:
The default value should also include all standard punctuation:
```
U+0020 (Space)
U+2000...U+206F (General Punctuation Block)
U+3000...U+303F (CJK Punctuation Block)
```

Reply:
> The default value should also include all standard punctuation:
> 
> ```
> U+0020 (Space)
> U+2000...U+206F (General Punctuation Block)
> U+3000...U+303F (CJK Punctuation Block)
> ```

These are the default ones copied from VSCode: https://github.com/godotengine/godot-proposals/issues/9837#issuecomment-2136118971

Underscore is also explicitly considered not as a punctuation: https://github.com/godotengine/godot/pull/46691 and https://github.com/godotengine/godot/pull/47604

Should I maybe then add the ones you wrote in a hardcoded way?

Reply:
I thought about it, but if I want to temporarily disable this behavior, clearing the whole list might be a bit overkill IMO. This may happen for example if:

* I do not use `snake_case` in the variables of my own project.
* Have "_" as an additional separator (assume that I have some more)
* Open someone else's project, who used snake case in variables, and navigate through their code.

If I do not have enable/disable function, I need to find underscore and delete it just for one project. Then add it again to make editor work as expected in my own project as separator.

Reply:
> I thought about it, but if I want to temporarily disable this behavior, clearing the whole list might be a bit overkill IMO. This may happen for example if:

This make sense, I guess we can simply add these ranges when reading `text_editor/behavior/navigation/use_custom_word_separators` value (or add extra `text_editor/behavior/navigation/use_default_word_separators` editor setting to control is default list is included or not, this way you can use only default/, only custom or both).